src/scanoss/file_filters.py

```diff
@@ -252,57 +260,85 @@ def __init__(
+                self.print_stderr(f'Running with {len(self.file_folder_pat_spec)} pattern filters.')
+            if self.size_pat_rules:
+                self.print_stderr(f'Running with {len(self.size_pat_rules)} size pattern rules.')
+            if self.skip_size:
+                self.print_stderr(f'Running with global skip size: {self.skip_size}')
+            if self.skip_extensions:
+                self.print_stderr(f'Running with extra global skip extensions: {self.skip_extensions}')
+            if self.skip_folders:
+                self.print_stderr(f'Running with extra global skip folders: {self.skip_folders}')
         all_files = []
         root_path = Path(root).resolve()
-
+        if not root_path.exists() or not root_path.is_dir():
+            self.print_stderr(f'ERROR: Specified root directory {root} does not exist or is not a directory.')
+            return all_files
+        # Walk the tree looking for files to process. While taking into account files/folders to skip
         for dirpath, dirnames, filenames in os.walk(root_path):
             dirpath = Path(dirpath)
             rel_path = dirpath.relative_to(root_path)
+            if dirpath.is_symlink():  # TODO should we skip symlink folders?
+                self.print_msg(f'WARNING: Found symbolic link folder: {dirpath}')
 
-            if self._should_skip_dir(str(rel_path), operation_type):
+            if self._should_skip_dir(str(rel_path)):  # Current directory should be skipped
                 dirnames.clear()
                 continue
-
             for filename in filenames:
                 file_path = dirpath / filename
                 all_files.append(str(file_path))
+        # End os.walk loop
+        # Now filter the files and return the reduced list
+        return self.get_filtered_files_from_files(all_files, str(root_path))
 
-        return self.get_filtered_files_from_files(all_files, operation_type, str(root_path))
-
-    def get_filtered_files_from_files(self, files: List[str], operation_type: str, scan_root: str = None) -> List[str]:
-        """Retrieve a list of files to scan or fingerprint from a given list of files based on filter settings.
+    def get_filtered_files_from_files(self, files: List[str], scan_root: str = None) -> List[str]:
+        """
+        Retrieve a list of files to scan or fingerprint from a given list of files based on filter settings.
 
         Args:
             files (List[str]): List of files to scan or fingerprint
-            operation_type (str): Type of operation ('scanning' or 'fingerprinting')
             scan_root (str): Root directory to scan or fingerprint
 
         Returns:
             list[str]: Filtered list of files to scan or fingerprint
         """
         filtered_files = []
         for file_path in files:
-            if not os.path.isfile(file_path):
+            if not os.path.exists(file_path) or not os.path.isfile(file_path) or os.path.islink(file_path):
```

Comment:
we could remove the `.exists()` check since both `isfile` and `islink` already check for existance